### PR TITLE
add user agent header to boto3 client for aws components

### PR DESCRIPTION
**Pull request by surajkota**
- Added versioning to the user-agent header to fetch from [license file](https://github.com/kubeflow/pipelines/blob/master/components/aws/sagemaker/THIRD-PARTY-LICENSES.txt#L1) at rutime.

- Tested using the latest image on mnist classification example pipeline

### CODE REVIEW

File: components/aws/sagemaker/common/_utils.py [hunk 1]
```diff
@@ -65,7 +66,10 @@ def add_default_client_arguments(parser):
 
 def get_sagemaker_client(region, endpoint_url=None):
     """Builds a client to the AWS SageMaker API."""
-    client = boto3.client('sagemaker', region_name=region, endpoint_url=endpoint_url)
+    session_config = botocore.config.Config(
+        user_agent='sagemaker-on-kubeflow-pipelines-v0.2'
```
Comment (Jeffwan) [inline components/aws/sagemaker/common/_utils.py:70]: Do you plan to update the version for every release?  is there more elegant way like you read from version file to build user agent?

Comment (surajkota) [reply components/aws/sagemaker/common/_utils.py]: updated to fetch at runtime

**Comment by RedbackThomson** [comment]
Can we add a unit test for this? It seems important enough to warrant its own test.

**Comment by surajkota** [comment]
Can take up test separately when we enhance unit tests for the components and in anycase header will keep propagating irrespective of the search being successful

**Pull request merged**